### PR TITLE
Refactored offichain worker code for both Thea and Ocex Pallets

### PR DESCRIPTION
**Pull request by zktony**
## Describe your changes
Refactored offichain worker code for both Thea and Ocex Pallets.

### CODE REVIEW

File: pallets/ocex/src/aggregator.rs
Comment (serhii-temchenko) [inline pallets/ocex/src/aggregator.rs:1]: Please add a module level and all public api (structs and methods) documentation

File: pallets/ocex/src/validator.rs [hunk 1]
```diff
@@ -130,9 +110,9 @@ impl<T: Config> Pallet<T> {
 			}
 		}
 
-		// Load the next ObMessages
+		// Load the next ObMessages¡
```
Comment (serhii-temchenko) [inline pallets/ocex/src/validator.rs:140]: What is this change about?

File: pallets/thea/src/aggregator.rs
Comment (serhii-temchenko) [inline pallets/thea/src/aggregator.rs:1]: Please add a module level and all public api (structs and methods) documentation

File: pallets/thea/src/resolver.rs
Comment (serhii-temchenko) [inline pallets/thea/src/resolver.rs:1]: Please add a module level and all public api (structs and methods) documentation

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/rpc/src/lib.rs [hunk 2]
```diff
@@ -124,21 +142,48 @@ where
...
+		at: Option<<Block as BlockT>::Hash>,
+	) -> RpcResult<ObCheckpoint> {
+		self.deny_unsafe.check_if_safe()?;
+		let api = self.client.runtime_api();
+		let at = match at {
+			Some(at) => at,
+			None => self.client.info().best_hash,
+		};
+		let offchain_storage = offchain::OffchainStorageAdapter::new(self.storage.clone());
+		if !offchain_storage.acquire_offchain_lock(3).await {
```
Comment (Gauthamastro) [inline pallets/ocex/rpc/src/lib.rs]: Retries must be defined as a constant

File: pallets/thea/src/validation.rs [hunk 3]
```diff
@@ -35,24 +24,35 @@ impl<T: Config> Pallet<T> {
...
-			//		c. Compute who should sign this and if its us then sign the payload
+			let next_incoming_message = AggregatorClient::<u64, T>::get_payload_for_nonce(
+				next_incoming_nonce,
+				network,
+				Destination::Parachain,
+			); //TODO: Replace with Parachain Struct object
+			let next_outgoing_message = AggregatorClient::<u64, T>::get_payload_for_nonce(
+				next_outgoing_nonce,
+				network,
+				Destination::Solochain,
```
Comment (Gauthamastro) [inline pallets/thea/src/validation.rs:58]: Resolve the TODO comments

File: primitives/orderbook/src/lib.rs [hunk 4]
```diff
@@ -125,3 +128,47 @@ impl<AccountId: Clone + Codec> SnapshotSummary<AccountId> {
...
+	}
+
+	/// Convert `ObCheckpointRaw` to `ObCheckpoint`.
+	/// # Returns
+	/// * `ObCheckpoint`: A new `ObCheckpoint` instance.
+	#[cfg(feature = "std")]
+	pub fn to_checkpoint(&self) -> ObCheckpoint {
+		ObCheckpoint {
+			snapshot_id: self.snapshot_id,
+			balances: self.balances.clone(),
```
Comment (Gauthamastro) [inline primitives/orderbook/src/lib.rs]: We can avoid clone by moving self too.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/ocex/rpc/src/offchain.rs
Comment (serhii-temchenko) [inline pallets/ocex/rpc/src/offchain.rs:1]: Please add license and module level documentation

File: pallets/thea/src/aggregator.rs
Comment (serhii-temchenko) [inline pallets/thea/src/aggregator.rs:1]: Please add license and module level documentation

File: pallets/thea/src/resolver.rs
Comment (serhii-temchenko) [inline pallets/thea/src/resolver.rs:1]: Please add license and module level documentation

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]
Please fix the CI errors

**Review by Gauthamastro — APPROVED** [review]

**Review by serhii-temchenko — APPROVED** [review]

**Pull request merged**